keras_core/random/seed_generator.py

```diff
@@ -52,49 +48,56 @@ def __init__(self, seed=None, **kwargs):
         else:
             self.backend = backend
 
-        self._initial_seed = seed
         if seed is None:
-            seed = make_default_seed()
-
-        if not isinstance(seed, int):
-            raise ValueError(
-                "Argument `seed` must be an integer. " f"Received: seed={seed}"
-            )
+            rng, seed = backend.random.make_default_seed()
+        else:
+            rng, seed = backend.random.make_initial_seed(seed)
 
-        def seed_initializer(*args, **kwargs):
-            dtype = kwargs.get("dtype", None)
-            return self.backend.convert_to_tensor([seed, 0], dtype=dtype)
+        if backend.backend() == "tensorflow":
```

Comment:
Why is this needed? Can't we use `uint32` everywhere?

Reply:
TF random generator use int32/int64, and with XLA only int32 is allowed. Casting it to uint32 can cause some unintended consequences, and will be hard to debug